packages/sanity/src/core/studio/components/navbar/StudioNavbar.tsx

```diff
@@ -139,123 +141,126 @@ export function StudioNavbar() {
             )}
 
-            {shouldRender.workspaces && (
-              <Box marginRight={2}>
-                <WorkspaceMenuButton collapsed />
-              </Box>
-            )}
-
-            <Box marginRight={shouldRender.brandingCenter ? undefined : 2}>
-              <NewDocumentButton
-                {...newDocumentOptions}
-                modal={shouldRender.brandingCenter ? 'dialog' : 'popover'}
-              />
-            </Box>
-
-            {/* Search */}
-            <LayerProvider>
-              <SearchProvider fullscreen={shouldRender.searchFullscreen}>
-                <BoundaryElementProvider element={document.body}>
-                  <PortalProvider element={searchFullscreenPortalEl}>
-                    {shouldRender.searchFullscreen && (
-                      <SearchDialog
-                        onClose={handleCloseSearchFullscreen}
-                        onOpen={handleOpenSearchFullscreen}
-                        open={searchFullscreenOpen}
-                      />
-                    )}
-                  </PortalProvider>
-                  {!shouldRender.searchFullscreen && <SearchField />}
-                </BoundaryElementProvider>
-              </SearchProvider>
-            </LayerProvider>
-
-            {shouldRender.tools && (
-              <Card flex={1} marginX={2} overflow="visible" paddingRight={1}>
-                <ToolMenu
-                  activeToolName={activeToolName}
-                  closeSidebar={handleCloseDrawer}
-                  context="topbar"
-                  isSidebarOpen={false}
-                  tools={tools}
-                />
-              </Card>
-            )}
-          </LeftFlex>
-
-          {shouldRender.brandingCenter && (
-            <Box marginX={1}>
-              <LogoButton href={rootHref} onClick={handleRootClick} title={title}>
-                <Logo title={title} />
-              </LogoButton>
-            </Box>
-          )}
-
-          <Flex gap={2}>
-            {(shouldRender.configIssues || shouldRender.resources) && (
-              <Card borderRight>
-                <Flex gap={1} paddingX={2}>
-                  {shouldRender.configIssues && <ConfigIssuesButton />}
-                  {shouldRender.resources && <ResourcesButton />}
-                </Flex>
-              </Card>
-            )}
-
-            <Flex align="center" gap={1}>
-              <PresenceMenu collapse={shouldRender.collapsedPresenceMenu} />
-              {shouldRender.tools && <UserMenu />}
-              {shouldRender.searchFullscreen && (
-                <Button
-                  aria-label={t('search.action-open-aria-label')}
-                  icon={SearchIcon}
-                  mode="bleed"
-                  onClick={handleOpenSearchFullscreen}
-                  ref={setSearchOpenButtonEl}
-                />
+            <Flex gap={2}>
+              {(shouldRender.configIssues || shouldRender.resources || shouldRender.tools) && (
```

Comment:
adds here the check for `shouldRender.tools` 
and below the line:
`packages/sanity/src/core/studio/components/navbar/NavDrawer.tsx`

These, in addition to adding the context are the only changes to this file.

Reply:
Looks good to me! The only thing I noticed is that things get a little cramped here now, resulting in some unhelpful string truncation.

## Before

<img width="1015" alt="before" src="https://github.com/sanity-io/sanity/assets/1454914/9e4dd31b-83fa-494c-aa68-6fd994385c1b">

## After

<img width="1015" alt="after" src="https://github.com/sanity-io/sanity/assets/1454914/82864fee-b152-4553-b099-44b5a8e80096">

Reply:
Nice catch, thanks, will keep a note to check that in case facelift doesn't goes out next week along with these changes. Facelift changes the behaviour of that navbar and gives more space.